### PR TITLE
chore(hardware): eeprom integ test

### PR DESCRIPTION
**Pull request by amitlissack**
# Overview

Improvement to the eeprom integration test

# Changelog

- eeprom test is parameterized to run against the nodes with an eeprom
- it is also parameterized to interact with multiple addresses with multiple data lengths

# Review requests

Not to be merged until [this](https://github.com/Opentrons/ot3-firmware/pull/364) is .

# Risk assessment

None

### CODE REVIEW

**Review by Laura-Danielle — APPROVED** [review]
nice

File: hardware/tests/firmware_integration/test_eeprom.py [hunk 1]
```diff
@@ -1,7 +1,10 @@
 """Tests for eeprom."""
 import asyncio
+from typing import Iterator
 
 import pytest
+from _pytest.fixtures import FixtureRequest
```
Comment (sfoster1) [inline hardware/tests/firmware_integration/test_eeprom.py:6]: what's up with the leading underscore?

**Review by sfoster1 — APPROVED** [review]
Nice, confused by the underscore tho

File: hardware/tests/firmware_integration/test_eeprom.py (hunk 1 above)
Comment (amitlissack) [reply hardware/tests/firmware_integration/test_eeprom.py:6]: `FixtureRequest` is an internal type. Pytest installs all its guts in `_pytest` for some unknown reason.

**Pull request merged**